js/event_target.ts

```diff
@@ -525,31 +550,11 @@ export class EventTarget implements domTypes.EventTarget {
       slotInClosedTree
     });
   }
-
-  get [Symbol.toStringTag](): string {
-    return "EventTarget";
-  }
-}
+};
 
 /** Built-in objects providing `get` methods for our
  * interceptable JavaScript operations.
  */
-Reflect.defineProperty(EventTarget.prototype, "host", {
-  enumerable: true,
-  writable: true
-});
-Reflect.defineProperty(EventTarget.prototype, "listeners", {
-  enumerable: true,
-  writable: true
-});
-Reflect.defineProperty(EventTarget.prototype, "mode", {
-  enumerable: true,
-  writable: true
-});
-Reflect.defineProperty(EventTarget.prototype, "nodeType", {
-  enumerable: true,
-  writable: true
-});
 Reflect.defineProperty(EventTarget.prototype, "addEventListener", {
```

Comment:
I'm having trouble understanding all the changes in event_target.ts - what is happening here?

Reply:
I think true public interfaces of EventTarget are only 3 methods (addEventListener, removeEventListener, and dispatchEvent), and the other methods and properties (host, mode, nodeType, etc) are implementation details for this specific implementation. Here we want to make `window` an EventTarget, but if we naively put all those non-public methods and properties to window, they all become global variables. To avoid that, I refactored EventTarget to use symbol-keyed properties (like this.host -> this[eventTargetHost]) and locally scoped helper methods (`eventTargetHelpers`). I think these tricks prevent leaking things to the global namespace.